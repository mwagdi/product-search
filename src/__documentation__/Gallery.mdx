---
name: Gallery
menu: Components
route: /components/Gallery
---

import { Playground, Props } from 'docz'
import Gallery from '../components/Gallery.js'
import './../assets/scss/app.scss'

# Gallery

## Props
<Props of={Gallery} />

## Basic usage
<Playground>
  <Gallery images={[
    "https://sspride.org/wp-content/uploads/2017/03/image-placeholder-500x500-300x300.jpg",
    "https://sspride.org/wp-content/uploads/2017/03/image-placeholder-500x500-300x300.jpg",
    "https://sspride.org/wp-content/uploads/2017/03/image-placeholder-500x500-300x300.jpg"
    ]} />
</Playground>
