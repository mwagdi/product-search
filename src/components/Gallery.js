import React,{ Component } from 'react';
import PropTypes from 'prop-types';

class Gallery extends Component{
    state = {
        current: 0
    }
    static propTypes = {
        /** Array of image URLs */
        images: PropTypes.array.isRequired
    }
    render(){
        const { images } = this.props;
        const { current } = this.state;
        return (
            <div className="gallery flex-container">
                <div className="gallery__viewer">
                    <img src={images[current]} alt="gallery view"/>
                </div>
                <div className="gallery__select flex-direction flex-column">
                    {images.map((image,i) => (
                        <div
                        onClick={() => this.setState({ current: i })}
                        key={i}
                        style={{backgroundImage: `url(${image})`}}
                        className={`gallery__select_item ${current === i ? 'active' : ''}`}></div>
                    ))}
                </div>
            </div>
        )
    }
}

export default Gallery;