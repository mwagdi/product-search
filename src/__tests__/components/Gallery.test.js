import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Gallery from '../../components/Gallery';

describe("Gallery component", () => {
    test('Renders Gallery component', () => {
        const images = [
            "dummy.jpg",
            "dummy1.jpg",
            "dummy2.jpg",
            "dummy3.jpg",
        ]
        const wrapper = shallow(<Gallery images={images} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
})