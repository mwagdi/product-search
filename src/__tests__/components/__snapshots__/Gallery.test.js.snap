// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component Renders Gallery component 1`] = `
<div
  className="gallery flex-container"
>
  <div
    className="gallery__viewer"
  >
    <img
      alt="gallery view"
      src="dummy.jpg"
    />
  </div>
  <div
    className="gallery__select flex-direction flex-column"
  >
    <div
      className="gallery__select_item active"
      key="0"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(dummy.jpg)",
        }
      }
    />
    <div
      className="gallery__select_item "
      key="1"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(dummy1.jpg)",
        }
      }
    />
    <div
      className="gallery__select_item "
      key="2"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(dummy2.jpg)",
        }
      }
    />
    <div
      className="gallery__select_item "
      key="3"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url(dummy3.jpg)",
        }
      }
    />
  </div>
</div>
`;
